File: Foundation/Arrays/reverse.cpp
#include<iostream>
#include<vector>
using namespace std;

void rotate(vector<int>& arr)
{
    int i = 0;
    int j = arr.size() - 1;
    while(i < j)
    {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;

        i++;
        j--;
    }
    for(int k = 0; k < arr.size(); k++)
    {
        cout<<arr[k];
    }
}

int main()
{
    int n;
    cin>>n;
    vector<int> arr(n);
    for(int i = 0; i < n; i++)
    {
        cin>>arr[i];
    }
    rotate(arr);
    return 0;
}

File: Foundation/Arrays/reverse.java
import java.util.Scanner;
public class rotate
{
    public static Scanner scn = new Scanner(System.in);
    public static void rotate(int[] arr)
    {
        int i = 0;
        int j = arr.length - 1;
        while(i < j)
        {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;

            i++;
            j--;
        }
        for(int k = 0; k < arr.length; k++)
        {
            System.out.print(arr[k];);
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nexInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++)
        {
            arr[i] = scn.nextInt();
        }
        rotate(arr);
    }
}